repo.js

```diff
@@ -1,81 +1,200 @@
-var inherits = require('util').inherits
-var pump = require('pump')
-var events = require('events')
-var swarm = require('webrtc-swarm')
-var Signalhub = require('signalhub')
-var hyperdrive = require('hyperdrive')
-var memdb = require('memdb')
-
-module.exports = Repo
-
-/**
- * A dat repository is a hyperdrive with some default settings.
- * @param {string} key    The key
- * @param {Object} opts   Options to use in the archive instance
- */
-function Repo (key, opts) {
-  if (!(this instanceof Repo)) return new Repo(key, opts)
-  var self = this
-  events.EventEmitter.call(this)
-  this.opts = opts || {}
-  this.db = this.opts.db || memdb()
-  this.drive = hyperdrive(this.db)
-  this.archive = this.drive.createArchive(key, this.opts)
-  this.key = this.archive.key.toString('hex')
-  this.privateKey = this.archive.privateKey
-  var signalhub = Signalhub('dat-' + this.archive.key.toString('hex'), opts.signalhub || 'https://signalhub.mafintosh.com')
-  this.swarm = this.swarm || swarm(signalhub)
-  self.join()
-  this._open(key)
-}
+const EventEmitter = require('events').EventEmitter
+const Signalhub = require('signalhubws')
+const hyperdrive = require('hyperdrive')
+const ram = require('random-access-memory')
+const websocket = require('websocket-stream')
+const WebrtcSwarm = require('webrtc-swarm')
+const pump = require('pump')
+var through = require('through2')
```

Comment:
Maybe should use consistent `const` here

Reply:
Yes! That's what I get for copy-pasting. 😅